akka-distributed-data/src/main/scala/akka/cluster/ddata/DurableStore.scala

```diff
@@ -115,26 +116,46 @@ final class LmdbDurableStore(config: Config) extends Actor with ActorLogging {
     case _     ⇒ config.getDuration("lmdb.write-behind-interval", MILLISECONDS).millis
   }
 
-  val env: Env[ByteBuffer] = {
-    val mapSize = config.getBytes("lmdb.map-size")
-    val dir = config.getString("lmdb.dir") match {
-      case path if path.endsWith("ddata") ⇒
-        new File(s"$path-${context.system.name}-${self.path.parent.name}-${Cluster(context.system).selfAddress.port.get}")
-      case path ⇒
-        new File(path)
-    }
-    log.info("Using durable data in LMDB directory [{}]", dir.getCanonicalPath)
-    dir.mkdirs()
-    Env.create()
-      .setMapSize(mapSize)
-      .setMaxDbs(1)
-      .open(dir, EnvFlags.MDB_NOLOCK)
+  val dir = config.getString("lmdb.dir") match {
+    case path if path.endsWith("ddata") ⇒
+      new File(s"$path-${context.system.name}-${self.path.parent.name}-${Cluster(context.system).selfAddress.port.get}")
+    case path ⇒
+      new File(path)
   }
+  log.info("Using durable data in LMDB directory [{}]", dir.getCanonicalPath)
+
+  // lazy init via `initDb`
+  private var env: Env[ByteBuffer] = null
+  // lazy init via `initDb`
+  private var db: Dbi[ByteBuffer] = null
+  // lazy init via `initDb`
+  private var keyBuffer: ByteBuffer = null
+  // lazy init via `initDb`
+  private var valueBuffer: ByteBuffer = null
```

Comment:
Wrap these in a class and have a single `Option[LmdbStuff]` instead of four nulls and a null-check?